Makefile

```diff
@@ -27,7 +29,7 @@ TOOLING=$(EMBEDMD_BIN) $(GOBINDATA_BIN) $(JB_BIN) $(GOJSONTOYAML) $(JSONNET_BIN)
 JSONNET_SRC=$(shell find ./jsonnet -type f -not -path "./jsonnet/vendor*")
 JSONNET_VENDOR=jsonnet/vendor
 
-GO_BUILD_RECIPE=GOOS=linux CGO_ENABLED=0 go build --ldflags="-s -X $(GO_PKG)/pkg/operator.Version=$(VERSION)"
+GO_BUILD_RECIPE=GOOS=$(GOOS) GOARCH=$(GOARCH) CGO_ENABLED=0 go build --ldflags="-s -X $(GO_PKG)/pkg/operator.Version=$(VERSION)"
```

Comment:
@dgrisonnet @paulfantom @simonpasquier  any thoughts on bringing this back which was deleted (accidentally?? 🤔 )  in 
* https://github.com/openshift/cluster-monitoring-operator/commit/84e101dab4f92ca886567f6303d476b74f930edb
* https://github.com/openshift/cluster-monitoring-operator/pull/960/files

Reply:
Seems accidental. Let's add it back.

Reply:
I don't have opinion on this, but I am also using Linux, so :shrug: Bear in mind than the same build recipe is used to build a binary that is later put into a container. That container needs to be linux one as even macOS is spawning linux virtual machine to run containers.

Reply:
Ah indeed, I hadn't thought about the local cmo use case when I changed the build recipe. As Pawel mentioned, it doesn't seem very meaningful to make the build cross-platform since CMO is only meant to be run inside of a container.

Reply:
This just makes it easier to run local-cmo on Mac. For container builds, it should work as  the script just relies on `go env` which should get the values right 🤞